src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+// Public. Tests if moving from `posPrev` to `posNext` will collide with a "wall"
+//`radius` is used to form a sphere for collision detection purposes
+//`checkHeight` is the positive height above posNext to perform certain checks
+//`actor` is the actor performing the check, allowing it to be skipped
+// returns true if a collision is detected, else false
+//`outPoly` returns the closest poly detected, while `outBgId` returns the poly owner
+s32 func_8003D52C(CollisionContext* colCtx, Vec3f* posResult, Vec3f* posNext, Vec3f* posPrev, f32 radius,
+                  CollisionPoly** outPoly, s32* outBgId, Actor* actor, f32 checkHeight) {
+    return func_8003CDD4(colCtx, COLPOLY_IGNORE_ENTITY, posResult, posNext, posPrev, radius, outPoly, outBgId, actor,
+                         checkHeight, 0);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042B2C.s")
+// Public. Tests if moving from `posPrev` to `posNext` will collide with a "wall"
+// Skips a check that occurs only when moving on the xz plane
+//`radius` is used to form a sphere for collision detection purposes
+//`checkHeight` is the positive height above posNext to perform certain checks
+//`actor` is the actor performing the check, allowing it to be skipped
+// returns true if a collision is detected, else false
+//`outPoly` returns the closest poly detected, while `outBgId` returns the poly owner
+s32 func_8003D594(CollisionContext* colCtx, Vec3f* posResult, Vec3f* posNext, Vec3f* posPrev, f32 radius,
+                  CollisionPoly** outPoly, s32* outBgId, Actor* actor, f32 checkHeight) {
+    return func_8003CDD4(colCtx, COLPOLY_IGNORE_ENTITY, posResult, posNext, posPrev, radius, outPoly, outBgId, actor,
+                         checkHeight, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042C3C.s")
+// Tests if a collision with a ceiling poly occurs with line `pos`.y to `pos`.y + `checkHeight`
+// `checkHeight` should be a positive value
+// returns true if a collision occurs, else false
+//`outPoly` returns the poly collided with, while `outBgId` returns the owner of the poly
+// `outY` returns the y coordinate of pos needed to not collide with `outPoly`
+s32 func_8003D600(CollisionContext* colCtx, u16 xpFlags, f32* outY, Vec3f* pos, f32 checkHeight,
+                  CollisionPoly** outPoly, s32* outBgId, Actor* actor) {
+    Lookup* lookupTbl;
+    Lookup* lookup;
+    s32 result;
+    Vec3f posTemp;
+    f32 tempY;
+
+    *outBgId = BGCHECK_SCENE;
+    *outY = pos->y;
+    if (BgCheck_PosErrorCheck(pos, "../z_bgcheck.c", 5206) == true) {
+        if (actor != NULL) {
+            osSyncPrintf("こいつ,pself_actor->name %d\n", actor->id);
+        }
+    }
+    lookupTbl = colCtx->lookupTbl;
+    if (func_8003C55C(colCtx, pos) == false) {
+        return false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042CB8.s")
+    lookup = func_8003AC54(colCtx, lookupTbl, pos);
```

Comment:
```suggestion
    subdivLookup = func_8003AC54(colCtx, lookupTbl, pos);
```
again, for clarity without function names